customer/types.go

```diff
@@ -0,0 +1,26 @@
+package customer
+
+import (
+	"sync"
+
+	"github.com/RTradeLtd/database/models"
+	"github.com/RTradeLtd/rtfs"
+)
+
+// Manager is used to handle managing customer objects
+type Manager struct {
+	um    *models.UserManager
+	ipfs  rtfs.Manager
+	mutex sync.Mutex
```

Comment:
is this used yet?

Reply:
the mutex or the customer object manager? Originally I was intending for customer object manager to be released with this PR, but it ended up being a bit more work then I expected so I want to target releasing this functionality in the next release after this one. So technically the customer object manager isn't being used yet. Mutex is being used when running `Manager::Update`

Reply:
We should not be putting core functions of a class in separate files from the definition

Reply:
ie as much as possible, `Manager`'s public functions should be in this file where `Manager` is defined

Reply:
that is a good point :ok_hand: